### PR TITLE
Move DashboardController to Jellyfin.Api

### PR DESCRIPTION
**Pull request by Ullmie02**
This moves DashboardController to Jellyfin.Api and removes the MediaBrowser.WebDashboard project and all its references.

### CODE REVIEW

File: Jellyfin.Api/Controllers/DashboardController.cs [hunk 1]
```diff
@@ -0,0 +1,264 @@
...
+            if (pages == null)
+            {
+                return NotFound(unavailableMessage);
+            }
+
+            // Don't allow a failing plugin to fail them all
+            var configPages = pages.Select(p =>
+                {
+                    return new ConfigurationPageInfo(p);
+                })
```
Comment (crobibero) [inline Jellyfin.Api/Controllers/DashboardController.cs:83]: Why change from DashboardService?

File: Jellyfin.Api/Controllers/DashboardController.cs [hunk 2]
```diff
@@ -0,0 +1,264 @@
...
+        private IEnumerable<ConfigurationPageInfo> GetConfigPages(IPlugin plugin)
+        {
+            return GetPluginPages(plugin).Select(i => new ConfigurationPageInfo(plugin, i.Item1));
+        }
+
+        private IEnumerable<Tuple<PluginPageInfo, IPlugin>> GetPluginPages(IPlugin plugin)
+        {
+            var hasConfig = plugin as IHasWebPages;
+
+            if (hasConfig == null)
```
Comment (crobibero) [inline Jellyfin.Api/Controllers/DashboardController.cs:251]:
```suggestion
            if (!(plugin is IHasWebPages hasConfig))
```

File: Jellyfin.Api/Controllers/DashboardController.cs [hunk 3]
```diff
@@ -0,0 +1,264 @@
...
+
+            // Bounce them to the startup wizard if it hasn't been completed yet
+            if (!_serverConfigurationManager.Configuration.IsStartupWizardCompleted
+                && !Request.Path.Value.Contains("wizard", StringComparison.OrdinalIgnoreCase)
+                && Request.Path.Value.Contains("index", StringComparison.OrdinalIgnoreCase))
+            {
+                return Redirect("index.html?start=wizard#!/wizardstart.html");
+            }
+
+            var stream = new FileStream(_resourceFileManager.GetResourcePath(basePath, path), FileMode.Open, FileAccess.Read);
```
[the remark this answers is not in the document]
Comment (crobibero) [reply Jellyfin.Api/Controllers/DashboardController.cs:204]: not needed, `FileStreamResult` disposes the stream properly

File: Jellyfin.Api/Controllers/DashboardController.cs (hunk 1 above)
Comment (Ullmie02) [reply Jellyfin.Api/Controllers/DashboardController.cs:83]: I thought this catch exception was covered by the exception middleware, so we can remove it

Comment (crobibero) [reply Jellyfin.Api/Controllers/DashboardController.cs:83]: If an exception is thrown in the select, the entire select fails. I think the idea was to return as many pages as possible, ignoring the ones that throw exception

Comment (Ullmie02) [reply Jellyfin.Api/Controllers/DashboardController.cs:83]: This makes sense. I will change it.

File: Jellyfin.Api/Controllers/DashboardController.cs [hunk 4]
```diff
@@ -0,0 +1,275 @@
...
+        }
+
+        private IEnumerable<Tuple<PluginPageInfo, IPlugin>> GetPluginPages(IPlugin plugin)
+        {
+            if (!(plugin is IHasWebPages))
+            {
+                return new List<Tuple<PluginPageInfo, IPlugin>>();
+            }
+
+            return (plugin as IHasWebPages)!.GetPages().Select(i => new Tuple<PluginPageInfo, IPlugin>(i, plugin));
```
Comment (crobibero) [inline Jellyfin.Api/Controllers/DashboardController.cs:267]:
```suggestion
            if (!(plugin is IHasWebPages hasWebPages))
            {
                return new List<Tuple<PluginPageInfo, IPlugin>>();
            }

            return hasWebPages.GetPages().Select(i => new Tuple<PluginPageInfo, IPlugin>(i, plugin));
```

**Pull request merged**